content/1_docs/2_cookbook/3_templating/0_understanding-oop/recipe.txt

```diff
@@ -0,0 +1,462 @@
+    }
+
+    public function getPublisher(): string {
+        return $this->publisher;
+    }
+    
+    public function getIsbn(): string {
+        return $this->isbn;
+    }
+      
+    public function getType(): string {
+        return $this->type;
+    }
+
+    public function setTitle(string $title): void {
+        $this->title = $title;;
+    }
+
+    public function setAuthor(string $author): void
+    {
+        $this->author = $author;
+    }
+
+    public function setPrice(float $price): void {
+        $this->price = $price;
+    }
+
+    public function setPublisher(string $publisher): void {
+        $this->publisher = $publisher;
+    }
+    
+    public function setIsbn(string $isbn): void {
+        $this->isbn = $isbn;
+    }
+
+    public function setType(string $type): void {
+        $this->type = $type;
+    }
+    
+    public function isEbook(): bool {
+        return $this->type === 'ebook';
+    }
+}
+```
+If you find books boring, create a similar class for games, films, smartphones or whatever you prefer.
+
+Now for some explanations…
+
+## Describing a class
+
+For starters, we define our new `Book` class with the keyword `class`. Inside the class, we define the properties that objects of this class should have, like a title, an author, a publisher etc. Of course, a book could have many more properties, and if you feel like it, you can extend this example yourself.
+
+### Properties and methods
+
+#### Visibility
+
+Class properties (variables and constants) and methods can have different (link: https://www.php.net/manual/en/language.oop5.visibility.php text: visibility statuses): private, protected or public. 
+
+We define a protected property with `protected` followed by the variable for the property, e.g. `protected $author`. The properties of a class usually should not be public to prevent users from assigning unexpected values.
+
+Public properties and methods like the setters and getters in our `Book` class can be called anywhere, not only inside the class where they are defined, but also inside other classes or scripts when we work with objects of that class.
+
+Protected properties and methods can be called within the defining class itself or within the children of that class, but not outside of these classes.
+
+Private properties and methods are even more restricted, they can only be used inside the defining class itself, not anywhere else.
+
+#### Getters and setters
+
+Because the properties are not public, we cannot access them directly, but have to create so called *getters* that return the value of a property (`getPrice()`, `getTitle()` etc.) and *setters*, which assign a value to the property of an object (`setPrice()`, `setTitle()` etc.). Getters and setters are often prefixed with `get-`/`set-` to indicate their purpose, but this is not required and you can name them what you want (although it is good practice to give methods, variables etc. speaking names and not name them foo or bar);
```

Comment:
```suggestion
Because the properties are not public, we cannot access them directly, but have to create so called *getters* that return the value of a property (`getPrice()`, `getTitle()` etc.) and *setters*, which assign a value to the property of an object (`setPrice()`, `setTitle()` etc.). Getters and setters are often prefixed with `get-`/`set-` to indicate their purpose, but this is not required and you can name them what you want (although it is good practice to give methods, variables etc. speaking names and not name them foo or bar).
```